plugin-server/tests/worker/vm/capabilities.test.ts

```diff
@@ -1,60 +1,148 @@
+                    { ingestion: true, processAsyncHandlers: true, processJobs: true, pluginScheduledTasks: true },
+                    {}
+                )
+                expect(shouldSetupPlugin).toEqual(false)
+            })
+        })
+
+        describe('ingestion', () => {
+            it('returns true if plugin has processEvent method and server has ingestion capability', () => {
+                const shouldSetupPlugin = shouldSetupPluginInServer({ ingestion: true }, { methods: ['processEvent'] })
+                expect(shouldSetupPlugin).toEqual(true)
+            })
+
+            it('returns false if plugin does not have processEvent method and server only has ingestion capability', () => {
+                const shouldSetupPlugin = shouldSetupPluginInServer(
+                    { ingestion: true },
+                    {
+                        methods: ['onEvent', 'onSnapshot', 'exportEvents'],
+                        scheduled_tasks: ['runEveryMinute'],
+                        jobs: ['someJob'],
+                    }
+                )
+                expect(shouldSetupPlugin).toEqual(false)
+            })
+        })
+
+        describe('scheduled tasks', () => {
+            it('returns true if plugin has any scheduled tasks and the server has pluginScheduledTasks capability', () => {
+                const shouldSetupPlugin = shouldSetupPluginInServer(
+                    { pluginScheduledTasks: true },
+                    { scheduled_tasks: ['runEveryMinute'] }
+                )
+                expect(shouldSetupPlugin).toEqual(true)
+            })
+
+            it('returns false if plugin has no scheduled tasks and the server has only pluginScheduledTasks capability', () => {
+                const shouldSetupPlugin = shouldSetupPluginInServer(
+                    { pluginScheduledTasks: true },
+                    { scheduled_tasks: [] }
+                )
+                expect(shouldSetupPlugin).toEqual(false)
+            })
+        })
+
+        describe('jobs', () => {
+            it('returns true if plugin has any jobs and the server has processJobs capability', () => {
+                const shouldSetupPlugin = shouldSetupPluginInServer({ processJobs: true }, { jobs: ['someJob'] })
+                expect(shouldSetupPlugin).toEqual(true)
+            })
+
+            it('returns false if plugin has no jobs and the server has only processJobs capability', () => {
+                const shouldSetupPlugin = shouldSetupPluginInServer({ processJobs: true }, { jobs: [] })
+                expect(shouldSetupPlugin).toEqual(false)
+            })
+        })
+
+        describe('processAsyncHandlers', () => {
+            it('returns true if plugin has onEvent, onSnapshot, onAction, or exportEvents and the server has processAsyncHandlers capability', () => {
+                for (const pluginMethod of ['onAction', 'onSnapshot', 'onEvent', 'exportEvents']) {
```

Comment:
Suggestion: use `it.each(['onAction', 'onSnapshot', 'onEvent', 'exportEvents'])` here: https://jestjs.io/docs/api#testeachtablename-fn-timeout

Looping and other meta-constructs are smelly in tests.